specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-09-18/kusto.json

```diff
@@ -0,0 +1,4351 @@
+        }
+      ],
+      "description": "Class representing an iot hub data connection.",
+      "x-ms-discriminator-value": "IotHub"
+    },
+    "EventGridDataConnection": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "The properties of the Event Grid data connection.",
+          "$ref": "#/definitions/EventGridConnectionProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataConnection"
+        }
+      ],
+      "description": "Class representing an Event Grid data connection.",
+      "x-ms-discriminator-value": "EventGrid"
+    },
+    "DataConnectionValidationListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataConnectionValidationResult"
+          },
+          "description": "The list of Kusto data connection validation errors."
+        }
+      },
+      "description": "The list Kusto data connection validation result."
+    },
+    "DataConnectionListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataConnection"
+          },
+          "description": "The list of Kusto data connections."
+        }
+      },
+      "description": "The list Kusto data connections operation response."
+    },
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody",
+          "description": "An error response from Kusto."
+        }
+      },
+      "description": "An error response from Kusto."
+    },
+    "ClusterCheckNameRequest": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Cluster name."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Kusto/clusters"
+          ],
+          "x-ms-enum": {
+            "name": "Type",
+            "modelAsString": false
+          },
+          "description": "The type of resource, Microsoft.Kusto/clusters."
+        }
+      },
+      "required": [
+        "name",
+        "type"
+      ],
+      "description": "The result returned from a cluster check name availability request."
+    },
+    "CheckNameRequest": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Kusto/clusters/databases",
+            "Microsoft.Kusto/clusters/attachedDatabaseConfigurations"
+          ],
+          "x-ms-enum": {
+            "name": "Type",
+            "modelAsString": false
+          },
+          "description": "The type of resource, for instance Microsoft.Kusto/clusters/databases."
+        }

```

Comment:
These different definition of same "Type" enum will likely cause problem in SDK.